File: src/main/java/fr/efrei/Calculator.java
package fr.efrei;

public class Calculator {
  private final Authorizer authorizer;

  Calculator(Authorizer authorizer) {
    this.authorizer = authorizer;
  }

  public int divide(int numerator, int denominator) {
    var authorized = authorizer.authorize();
    if (authorized) { // <- bug here, should be if (!authorized)
      throw new UnauthorizedAccessException("Not authorized");
    }
    return numerator / denominator;
  }
}

File: cs/Doppelganger/Calculator.cs
namespace Doppelganger
{
    public class Calculator
    {
        private readonly IAuthorizer _authorizer;

        public Calculator(IAuthorizer authorizer)
        {
            _authorizer = authorizer;
        }

        public int Add(int x, int y)
        {
            bool authorized = _authorizer.Authorize();
            // BUG! Should be if(!authorized)
            if (authorized)
            {
                throw new Exception("Not authorized");
            }
            return x + y;
        }
    }
}
